Simplenote/Classes/SignupRemote.swift

```diff
@@ -2,53 +2,14 @@ import Foundation
 
 // MARK: - SignupRemote
 //
-class SignupRemote {
-    enum Result: Equatable {
-        static func == (lhs: SignupRemote.Result, rhs: SignupRemote.Result) -> Bool {
-            switch (lhs, rhs) {
-            case (.success, .success):
-                return true
-            case (.failure(let code1, _), .failure(let code2, _)):
-                return code1 == code2
-            default:
-                return false
-            }
-
-        }
-
-        case success
-        case failure(_ statusCode: Int, _ error: Error?)
-    }
-
-    private let urlSession: URLSession
-
-    init(urlSession: URLSession = URLSession.shared) {
-        self.urlSession = urlSession
-    }
-
-    /// Send signup request for specified email address
-    ///
-    func signup(with email: String, completion: @escaping (_ result: Result) -> Void) {
+class SignupRemote: Remote {
+    func signup(with email: String, completion: @escaping (_ result: Result<Data?, RemoteError>) -> Void) {
         guard let request = request(with: email) else {
-            completion(.failure(0, nil))
+            completion(.failure(RemoteError.network))
```

Comment:
Note that this isn't really a network error (but a URLRequest that couldn't be built)